From a97cc87dc8b115be60980db18ecffcd07802a534 Mon Sep 17 00:00:00 2001
From: Joshua Yan <52796499+yanjos-dev@users.noreply.github.com>
Date: Fri, 5 Jun 2020 12:29:41 -0700
Subject: [PATCH 4/5] Disable optimization for TI compiler v18 or higher

* Disable the TI ARM compiler v18 or higher due to some issues
  with the intrinsics.
---
 src/compilersupport_p.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/compilersupport_p.h b/src/compilersupport_p.h
index 81689dd..b124272 100644
--- a/src/compilersupport_p.h
+++ b/src/compilersupport_p.h
@@ -92,6 +92,10 @@
 #  define __has_builtin(x)  0
 #endif
 
+
+/* Disable this optimization for TI ARM compiler v18 or higher because it has some issues
+with these intrinsics. */
+#if !defined(__TI_COMPILER_VERSION__) || __TI_COMPILER_VERSION__ < 18000000
 #if (defined(__GNUC__) && (__GNUC__ * 100 + __GNUC_MINOR__ >= 403)) || \
     (__has_builtin(__builtin_bswap64) && __has_builtin(__builtin_bswap32))
 #  if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
@@ -146,6 +150,7 @@
 #    define cbor_htons
 #  endif
 #endif
+#endif
 #ifndef cbor_ntohs
 #   define cbor_ntohs(x) (((uint16_t)x >> 8) | ((uint16_t)x << 8))
 #   define cbor_htons    cbor_ntohs
@@ -226,4 +231,3 @@ static inline bool add_check_overflow(size_t v1, size_t v2, size_t *r)
 }
 
 #endif /* COMPILERSUPPORT_H */
-
-- 
2.22.0.windows.1

